app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -245,7 +243,7 @@ def server_status
     def facts
       User.as_anonymous_admin do
         sync_task(::Actions::Katello::Host::Update, @host, rhsm_params)
-        @host.subscription_facet.update_facts(rhsm_params[:facts]) unless rhsm_params[:facts].blank?
+        Katello::Host::SubscriptionFacet.update_facts(@host, rhsm_params[:facts])
```

Comment:
Actually is it saved to the db as part of the fact importer, if any error occurs within the plan() phase, it rolls it all back.